src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSArgumentSetterConfig.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2020 AdaptiveScale, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.gcs.actions;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.gcp.bigquery.source.BigQuerySource;
+import io.cdap.plugin.gcp.common.GCPReferenceSourceConfig;
+import io.cdap.plugin.gcp.gcs.GCSPath;
+
+/**
+ * Holds configuration required for configuring {@link BigQuerySource}.
+ */
+public final class GCSArgumentSetterConfig extends GCPReferenceSourceConfig {
```

Comment:
we should probably not use referencesourceconfig here, since actions don't have a reference name. 

Also, can we make a change here to accept the contents of the service account JSON, instead of a path to the file?

Reply:
@bdmogal have a look , reference is removed and also the support for service account JSON is implemented.